template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -291,6 +311,52 @@ public String getStackVersion() {
         return cmTemplate.getCdhVersion();
     }
 
+    @Override
+    public Map<String, Map<String, ServiceAttributes>> getHostGroupBasedServiceAttributes() {
+        // The only attributes required at the moment relate to the YARN NodeManager.
+        // This functionality can be moved to an interface / separated if additional components
+        // need this functionality.
+        Map<String, Set<ServiceComponent>> hgToNonGwServiceComponents = getNonGatewayServicesByHostGroup();
+        Map<String, Set<ServiceComponent>> hgToNonGwServiceComponentsWithYarnNMs = new HashMap<>();
+        for (Entry<String, Set<ServiceComponent>> entry : hgToNonGwServiceComponents.entrySet()) {
+            for (ServiceComponent sc : entry.getValue()) {
+                if (sc.getService().equalsIgnoreCase(YarnRoles.YARN) && sc.getComponent()
+                        .equalsIgnoreCase(YarnRoles.NODEMANAGER)) {
+                    hgToNonGwServiceComponentsWithYarnNMs.put(entry.getKey(), entry.getValue());
+                    break;
+                }
+            }
+        }
+        // We have the hostGroups without GATEWAY components with YARN NodeManagers
+        Map<String, Set<String>> componentsByHostGroup = hgToNonGwServiceComponentsWithYarnNMs.entrySet()
+                .stream().collect(
+                        toMap(
+                                Entry::getKey,
+                                e -> e.getValue().stream()
+                                .map(ServiceComponent::getComponent)
+                                .collect(Collectors.toUnmodifiableSet())
+                ));
+        // Re-using the current LoadBasedAutoScaling recommendation along with the above YARN filter.
```

Comment:
this comment also seems unnecessary or I'm missing something

Reply:
Goign to modify this comment a bit. It is relevant in the sense that determining a node as 'compute' means that it can be AutoScaled, and so it is important to re-use the autoscale recommendation logic.